src/PrestaShopBundle/Resources/config/services/core/form/choice_provider.yml

```diff
@@ -12,17 +12,25 @@ services:
     arguments:
       - '@=service("prestashop.adapter.data_provider.language").getLanguages(false)'
 
-  prestashop.core.form.choice_provider.country_by_id:
-    class: 'PrestaShop\PrestaShop\Core\Form\ChoiceProvider\CountryByIdChoiceProvider'
+  PrestaShop\PrestaShop\Core\Form\ChoiceProvider\CountryByIdChoiceProvider:
+    public: false
+    autowire: true
     arguments:
       - '@=service("prestashop.adapter.legacy.context").getLanguage().id'
       - '@prestashop.adapter.data_provider.country'
 
-  prestashop.core.form.choice_provider.currency_by_id:
-    class: 'PrestaShop\PrestaShop\Core\Form\ChoiceProvider\CurrencyByIdChoiceProvider'
+  prestashop.core.form.choice_provider.country_by_id:
+    alias: PrestaShop\PrestaShop\Core\Form\ChoiceProvider\CountryByIdChoiceProvider
+
+  PrestaShop\PrestaShop\Core\Form\ChoiceProvider\CurrencyByIdChoiceProvider:
+    public: false
+    autowire: true
     arguments:
       - '@prestashop.adapter.data_provider.currency'
```

Comment:
If you add autowire, autowire arguments as well otherwise is is not useful. (applies for all services you edited)